### PR TITLE
feat(1983,1884): api-catalog-spa : implemented multi spec support

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Closes

1. 1983
2. 1884

# Explain the feature/fix

1. Implemented api catalog multiple specsheet support
2. Added advance validation for specsheets
3. Added  new tool redoc for open api documentation
4. Depreciated voyager
5. Migrated to vitejs
6. Added multi env subscription support

## Does this PR introduce a breaking change

Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

## API List Page

Screenshot 2022-05-12 at 7 10 23 PM

## API Details Page Rest

Screenshot 2022-05-12 at 7 10 28 PM

## API Details Page GQL

Screenshot 2022-05-12 at 7 10 32 PM

## ReDoc

Screenshot 2022-05-12 at 7 10 47 PM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

**Comment by akhilmhdh** [comment]
Added .test.env

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/APIReview/APIReview.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import { config } from 'config';
+import { CatalogBigButton, ReadMore } from 'components';
+import { FormData } from 'pages/APICudPage/APICudPage.types';
+
+import { Description } from './components/Description';
+
+const apiOptions: Record<string, { title: string; desc: string; image: string }> = {
+  REST: {
+    title: 'REST',
+    desc: 'Confirms to OpenAPI specs',
```
Comment (riginoommen) [inline packages/api-catalog-spa/src/pages/ApiCUDPage/components/APIReview/APIReview.tsx:33]:
```suggestion
    desc: 'API that conforms to the constraints of Representational state transfer architecture,
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Comment by riginoommen** [comment]
Build is failing

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/APIReview/APIReview.tsx (hunk 1 above)
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/pages/ApiCUDPage/components/APIReview/APIReview.tsx:33]: This was already discussed before. We took a short sentence like that to keep it consistent with the graphql one and also to keep it without breaking the design.

File: packages/api-catalog-spa/src/context/BreadcrumbContext/lookupTable.ts [hunk 2]
```diff
@@ -16,7 +16,23 @@ export const LOOK_UP_TABLE: Record<string, Breadcrumb> = {
...
+    label: 'Redoc Toolbox',
+    url: '/',
+  },
+  gqlPlayground: {
+    label: 'GraphQL Playground',
+    url: '/',
+  },
+  voyager: {
+    label: 'GraphQL Voyager',
+    url: '/',
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/context/BreadcrumbContext/lookupTable.ts]: Removed

File: packages/api-catalog-spa/src/context/RecentVisitContext/RecentVisitContext.tsx [hunk 3]
```diff
@@ -1,40 +1,80 @@
...
+    localStorage.removeItem(key);
+    return [];
+  }
+  return data.log;
+};
+
+const saveToLocalStorage = (log: RecentVisitLog[]) => {
+  const key = getKey();
+  const now = new Date();
+  const jsonLog = JSON.stringify({ log, expiry: now.getTime() + TTL });
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/context/RecentVisitContext/RecentVisitContext.tsx:51]: Answered below

File: packages/api-catalog-spa/package.json [hunk 4]
```diff
@@ -1,91 +1,71 @@
...
-    "dayjs": "^1.10.7",
+    "@patternfly/react-core": "^4.206.2",
+    "@patternfly/react-icons": "^4.57.2",
+    "@patternfly/react-styles": "^4.56.2",
+    "@sentry/react": "^6.19.6",
+    "@types/js-yaml": "^4.0.5",
+    "dayjs": "^1.11.1",
+    "graphql": "^16.4.0",
     "graphql-playground-react": "^1.7.28",
     "graphql-voyager": "^1.0.0-rc.31",
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/api-catalog-spa/package.json]: Removed

**Comment by akhilmhdh** [comment]
S

> Build is failing

Same ans. All those errors doesn't make sense to me. Everyone of them is typed with a type.

And its building fine
Screenshot 2022-05-13 at 12 00 32 AM

**Review by riginoommen — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**